core/src/debug_client.rs

```diff
@@ -0,0 +1,71 @@
+//! This module contains code for an http client that is used for the VSCode debug plugin.
+
+use prost::Message;
+use reqwest;
+use std::time::Duration;
+use temporal_sdk_core_protos::temporal::api::history::v1::History;
+
+const CLIENT_NAME: &str = "temporal-core";
+const CLIENT_VERSION: &str = "0.1.0";
+
+/// Struct representing the internal debug client
+#[derive(Clone)]
+pub struct DebugClient {
+    /// URL for the local instance of the debugger server
+    debugger_url: String,
+
+    /// reqwest::Client to avoid re-creating a client for every request.
```

Comment:
You never need to repeat type names in comments because the doc generation and any good IDE is gonna show you the type anyway
```suggestion
    /// Underlying HTTP client
```